core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -570,7 +565,7 @@ class BlockManagerMasterEndpoint(
       val externalShuffleServiceBlockStatus =
         if (externalShuffleServiceRddFetchEnabled) {
           val externalShuffleServiceBlocks = blockStatusByShuffleService
-            .getOrElseUpdate(externalShuffleServiceIdOnHost(id), new JHashMap[BlockId, BlockStatus])
+            .getOrElseUpdate(externalShuffleServiceIdOnHost(id), new BlockStatusPerBlockId)
```

Comment:
Seems like we never clear the key after this change. Could you add some comments (maybe here) to explain?

Reply:
done